AccessControl: Update permissions and scopes in docs (#35263)
grafana/grafana: docs/sources/enterprise/access-control/permissions.md

## Action definitions

The following list contains fine-grained access control actions.

Actions | Applicable scopes | Descriptions
--- | --- | ---
roles:list | roles:* | List available roles without permissions.
roles:read | roles:* | Read a specific role with it's permissions.
roles:write | permissions:delegate | Create or update a custom role.
roles:delete | permissions:delegate | Delete a custom role.
roles.builtin:list | roles:* | List built-in role assignments.
roles.builtin:add | permissions:delegate | Create a built-in role assignment.
roles.builtin:remove | permissions:delegate | Delete a built-in role assignment.

> reports.admin:create | reports:* | Create reports.

Author: Need to add this the concerned HTTP endpoint

Author: I cannot find the `POST /api/reports/new` So I assume we can skip that

Reviewer: This endpoint is not documented in the HTTP API in the reporting page it just mentioned you need to have the right permission. For now having this mentioned in list of actions is sufficient

users.authtoken:list | global:users:* | List authentication tokens that are assigned to a user.
users.authtoken:update | global:users:* | Update authentication tokens that are assigned to a user.
users.password:update | global:users:* | Update a user’s password.
users:delete | global:users:* | Delete a user.
users:create | n/a | Create a user.
users:enable | global:users:* | Enable a user.
users:disable | global:users:* | Disable a user.
users.permissions:update | global:users:* | Update a user’s organization-level permissions.
users:logout | global:users:* | Log out a user.
users.quotas:list | global:users:* | List a user’s quotas.
users.quotas:update | global:users:* | Update a user’s quotas.
org.users.read | users:* | Get user profiles within an organization.
org.users.add | users:* | Add a user to an organization.
org.users.remove | users:* | Remove a user from an organization.
org.users.role:update | users:* | Update the organization role (`Viewer`, `Editor`, `Admin`) for an organization.
ldap.user:read | n/a | Get a user via LDAP.
ldap.user:sync | n/a | Sync a user via LDAP.
ldap.status:read | n/a | Verify the LDAP servers’ availability.

> status:accesscontrol | service:access-control | Get access-control enabled status.

Author: Added this one. Need to double check we are ok to communicate about it. @vtorosyan ?

Author: We did advertise the enpoint in docs/source/http_api/access_control.md so I assume this is ok :)

Reviewer: yes this is needed

## Scope definitions

The following list contains fine-grained access control scopes.

Scopes | Descriptions
--- | ---

> roles:* | Restrict an action to a set of roles. For example, `roles:*` matches any role, `roles:randomuid` matches only the role with UID `randomuid` and `roles:custom:reports:{editor,viewer}` matches both `custom:reports:editor` and `custom:reports:viewer` roles.

Reviewer: I really like the examples in the scopes and the new descriptions 👍

> roles:* | Restrict an action to a set of roles. For example, `roles:*` matches any role, `roles:randomuid` matches only the role with UID `randomuid` and `roles:custom:reports:{editor,viewer}` matches both `custom:reports:editor` and `custom:reports:viewer` roles.

Reviewer 2: Those are not valid UIDs post https://github.com/grafana/grafana-enterprise/issues/1466

Author: Good point. We did not backport that change. But if we still need to change this, then conf/access-control/sample.yaml and docs/sources/enterprise/access-control/provisioning.md need to be updated

Reviewer 3: the {editor,viewer} ?

Reviewer: I'd recommend to keep it simple for now and give an example of restricting to one role UID.

Reviewer 2: Colons are not valid characters in UIDs post URI restrictions `:`. Good point about provisioning docs and samples, we'll have to fix that.

> roles:* | Restrict an action to a set of roles. For example, `roles:*` matches any role, `roles:randomuid` matches only the role with UID `randomuid` and `roles:custom:reports:{editor,viewer}` matches both `custom:reports:editor` and `custom:reports:viewer` roles.

Reviewer: ```suggestion
roles:* | Restrict an action to a set of roles. For example, `roles:*` matches any role, `roles:5vV9FheMk` matches only the role with UID `5vV9FheMk` and `roles:{PS49Fh6Gk,KHVrKheGz}` matches both roles with UID `PS49Fh6Gk` and `KHVrKheGz` .
```

> permissions:delegate | The scope is only applicable for roles associated with the Access Control itself and indicates that you can delegate your permissions only, or a subset of it, by creating a new role or making an assignment.

Author: This one is causing us trouble to rewrite

> permissions:delegate | The scope is only applicable for roles associated with the Access Control itself and indicates that you can delegate your permissions only, or a subset of it, by creating a new role or making an assignment.

Reviewer 3: ```suggestion
permissions:delegate | The scope is only applicable for roles associated with the Access Control itself and indicates that you can delegate your own permissions by creating a new role or making an assignment.
```

> permissions:delegate | The scope is only applicable for roles associated with the Access Control itself and indicates that you can delegate your permissions only, or a subset of it, by creating a new role or making an assignment.

Reviewer: ```suggestion
permissions:delegate | The scope is only applicable for the fine-grained access control itself. It indicates that you create a new role or a built-in role assignments with a subset of permissions that you have.
```

Reviewer 3: ```suggestion
permissions:delegate | This scope is only applicable for fine-grained access control. It allows you to create a new role or built-in role assignment with a subset of your own permissions.
```

Reviewer 3: Tweaked vardan's a bit

reports:* | Restrict an action to a set of reports. For example, `reports:*` matches any report and `reports:1` matches the report with id `1`.

> service:access-control | Restrict an action to target only the access-control service. For example, you can use this in conjunction with the `provisioning:reload` or the `status:accesscontrol` actions.

Reviewer: ```suggestion
service:accesscontrol | Restrict an action to target only the fine-grained access control service. For example, you can use this in conjunction with the `provisioning:reload` or the `status:accesscontrol` actions.
```

The `access-control` has been changed to `accesscontrol` for consistency.

> global:users:* | Restrict an action to a set of global users.

Author: Maybe add an example or two

> users:* | Restrict an action to a set of users from an organization.

Author: Examples
